applications/tari_app_utilities/src/network_check.rs

```diff
@@ -0,0 +1,70 @@
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use tari_common::{
+    configuration::Network,
+    exit_codes::{ExitCode, ExitError},
+};
+use tari_features::resolver::Target;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum NetworkCheckError {
+    #[error("The network {0} is invalid for this binary built for MainNet")]
+    MainNetBinary(Network),
+    #[error("The network {0} is invalid for this binary built for NextNet")]
+    NextNetBinary(Network),
+    #[error("The network {0} is invalid for this binary built for TestNet")]
+    TestNetBinary(Network),
+}
+
+impl From<NetworkCheckError> for ExitError {
+    fn from(err: NetworkCheckError) -> Self {
+        Self::new(ExitCode::NetworkError, err)
+    }
+}
+
+#[cfg(tari_network_mainnet)]
+pub const TARGET_NETWORK: Target = Target::MainNet;
+
+#[cfg(tari_network_nextnet)]
+pub const TARGET_NETWORK: Target = Target::NextNet;
+
+#[cfg(all(not(tari_network_mainnet), not(tari_network_nextnet)))]
+pub const TARGET_NETWORK: Target = Target::TestNet;
+
+pub fn is_network_choice_valid(network: Network) -> Result<(), NetworkCheckError> {
+    match TARGET_NETWORK {
+        Target::MainNet => match network {
+            Network::MainNet | Network::StageNet => Ok(()),
+            _ => Err(NetworkCheckError::MainNetBinary(network)),
+        },
+        Target::NextNet => match network {
+            Network::NextNet => Ok(()),
+            _ => Err(NetworkCheckError::NextNetBinary(network)),
+        },
+        Target::TestNet => match network {
+            Network::LocalNet | Network::Igor | Network::Esmeralda => Ok(()),
+            _ => Err(NetworkCheckError::TestNetBinary(network)),
+        },
+    }
```

Comment:
```suggestion
    match (TARGET_NETWORK, network) {
        (Target::MainNet, Network::MainNet | Network::StageNet) => Ok(()),
        (Target::MainNet,   _) => Err(NetworkCheckError::MainNetBinary(network)),
        (Target::NextNet,  Network::NextNet) => Ok(()),
        (Target::NextNet, _) => Err(NetworkCheckError::NextNetBinary(network)),
        (Target::TestNet,   Network::LocalNet | Network::Igor | Network::Esmeralda) => Ok(()),
        (Target::TestNet, _) => Err(NetworkCheckError::TestNetBinary(network)),
        },
    }
```

A lot more readable I think you'll find